net-libs/libwebsockets/libwebsockets-1.7.4.ebuild

```diff
@@ -0,0 +1,27 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+inherit cmake-utils
+
+DESCRIPTION="canonical libwebsockets.org websocket library"
+HOMEPAGE="https://libwebsockets.org/"
+SRC_URI="https://github.com/warmcat/libwebsockets/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="LGPL2"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="static-libs"
+
+DEPEND="
+	dev-util/cmake
+	dev-libs/openssl
+	"
+RDEPEND=""
+
+src_install() {
+	 cmake-utils_src_install
+	 use static-libs || rm -f "${D}"/usr/lib*/lib*.a || die
```

Comment:
Where possible, it's preferred to avoid building the static library in the first place. Fortunately, the root CMakeLists.txt has some options to do this.

I ended up with something like:

```
src_configure() {
	 local mycmakeargs=(
	 	 -DLWS_WITH_STATIC=$(usex static-libs)
	 	 -DLWS_LINK_TESTAPPS_DYNAMIC=$(usex !static-libs)e
        )

	 cmake-utils_src_configure
}
```

There's a few other interesting options in there too, like to control ipv6, openssl, zlib etc. support. It's up to you whether you think they're worth exposing or not.